pre_commit_dbt/check_macro_is_referenced.py

```diff
@@ -0,0 +1,66 @@
+import argparse
+from typing import Any
+from typing import Dict
+from typing import Optional
+from typing import Sequence
+
+from pre_commit_dbt.utils import add_filenames_args
+from pre_commit_dbt.utils import add_manifest_args
+from pre_commit_dbt.utils import get_json
+from pre_commit_dbt.utils import get_macro_sqls
+from pre_commit_dbt.utils import get_macros
+from pre_commit_dbt.utils import JsonOpenError
+
+def get_macro_name(macros_ref):
+    return map(lambda x:x.split(".")[-1], macros_ref) 
+
+def get_macros_in_models(manifest):
+    """
+    Gets macros listed in the "depends_on" key of the "nodes" key in the manifest
+    Arguments:
+        manifest: manifest file 
+    Returns:
+        macros: list of macros
+    """
```

Comment:
DISCLAIRMER: controversial topic but I'm not in favor of docstrings. Usually good type hinting and good naming is clear enough to understand what's going on in the function. I think your docstring here can be replaced by doing something like this:

```suggestion
def get_macros_in_models(manifest: Dict[str, <replace by the types of values>]) -> List[<replace by type of macros>]:
```

Let me know if it makes sense to you or not. Note that you'll have to import the new types like List from typing library.